src/rpc/mining.cpp

```diff
@@ -870,6 +870,134 @@ UniValue estimatesmartfee(const JSONRPCRequest& request)
 /* ************************************************************************** */
 /* Merge mining.  */
 
+/* The variables below are used to keep track of created and not yet
+       submitted auxpow blocks.  Lock them to be sure even for multiple
+       RPC threads running in parallel.  */
+static CCriticalSection cs_auxblockCache;
+static std::map<uint256, CBlock*> mapNewBlock;
+static std::vector<std::unique_ptr<CBlockTemplate>> vNewBlockTemplate;
+
+static 
+void _AuxMiningCheck()
```

Comment:
Please remove the underscore in the function names.  I don't think that names with an underscore are used anywhere in the code currently, so we should stay consistent to that.  (For me, just removing it and calling the function `AuxMiningCheck` would be fine.)